lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -6,32 +6,31 @@
 library js_promise;
 
 import 'dart:js_interop';
-
-import 'package:js/js_util.dart' as js_util;
-
-import '../util.dart';
-
-extension CallExtension on JSFunction {
-  external void call(JSAny? this_, JSAny? object);
-}
-
-@JS('Promise')
-external JSAny get _promiseConstructor;
-
-JSPromise<JSAny?> createPromise(JSFunction executor) =>
-  js_util.callConstructor(
-    _promiseConstructor,
-    <Object>[executor],
-  );
-
-
-JSPromise<JSAny?> futureToPromise<T extends JSAny?>(Future<T> future) {
-  return createPromise((JSFunction resolver, JSFunction rejecter) {
-    future.then(
-      (T value) => resolver.call(null, value),
-      onError: (Object? error) {
-        printWarning('Rejecting promise with error: $error');
-        rejecter.call(null, null);
+import 'dart:js_interop_unsafe';
+
+/// This is the same as package:js_interop's FutureToPromise (.toJS), but with
+/// a more descriptive error message.
+///
+/// TODO(srujzs/dit): Move to js_interop's .toJS.
```

Comment:
We need to revisit the JS Interop errors. The current version boxes hides the message and stack trace in a non-standard place, and makes the reporting in the JS console bad (I'll probably create an issue in dart-lang)